etc/kayobe/ansible/ubuntu-upgrade.yml

```diff
@@ -0,0 +1,112 @@
+---
+# To prevent Ansible role dependency errors, this playbook requires that environment variable
+# ANSIBLE_ROLES_PATH is defined and includes '$KAYOBE_PATH/ansible/roles' at seed-hypervisor.
+- name: Migrate hosts from Ubuntu Focal 20.04 to Jammy 22.04
+  hosts: overcloud:infra-vms:seed
+  vars:
+    ansible_python_interpreter: /usr/bin/python3
+  tasks:
+    - name: Assert that hosts are running Ubuntu Focal
+      assert:
+        that:
+          - ansible_facts.distribution == 'Ubuntu'
+          - ansible_facts.distribution_major_version == '20'
+          - ansible_facts.distribution_release == 'focal'
+          - os_distribution == 'ubuntu'
+        fail_msg: >-
+          This playbook is only designed for Ubuntu Focal 20.04 hosts. Ensure
+          that you are limiting it to only run on Focal hosts and
+          os_distribution in set to ubuntu.
+
+    - name: Ensure apt packages are up to date
+      apt:
+        update_cache: true
+        upgrade: yes
+      become: true
+
+    # NOTE: We cannot use apt_repository here because definitions must exist within the standard repos.list
+    - name: Ensure Jammy repo definitions exist in sources.list
+      blockinfile:
+        path: /etc/apt/sources.list
+        block: |
+          deb {{ stackhpc_repo_ubuntu_jammy_url }} jammy main restricted universe multiverse
+          deb {{ stackhpc_repo_ubuntu_jammy_url }} jammy-updates main restricted universe multiverse
+          deb {{ stackhpc_repo_ubuntu_jammy_url }} jammy-backports main restricted universe multiverse
+          deb {{ stackhpc_repo_ubuntu_jammy_security_url }} jammy-security main restricted universe multiverse
+      become: true
+
+    - name: Ensure do-release-upgrade is installed
+      package:
+        name: ubuntu-release-upgrader-core
+        state: latest
+      become: true
+
+    - name: Print selection of Deities to pray to
+      debug:
+        msg: |
+          The following Deities are associated with luck or good fortune:
+            Hamingja - Norse guardian spirit
+            Fortuna - Roman goddess of fortune and luck
+            Tyche - Greek goddess of success, fortune, Luck, and prosperity
+            Nang Kwak - Thai folklore goddess deemed to bring good fortune and prosperity
```

Comment:
```suggestion
```

This repository is synced to and forked by multiple clients. We need to maintain our professionalism when doing so. This shouldn't be in here.

Reply:
I'd say it's harmless fun, but if there's a concensus that it should be removed then we can get rid of it

Reply:
Maybe @markgoddard or @oneswig can weigh in

Reply:
Does that mean the procedure is unreliable and you really need to pray? ;-)

Reply:
It's pretty reliable, but if it fails then it's just a big pain

Reply:
Big pain == reprovision the host? Maybe it's easier just to reprovision the host with Jammy image?

Reply:
In-place has been better than reprovision. It's relatively fast and consistent. Maybe worth adding reprovision as a backup option to the guide though